python/cucim/src/cucim/skimage/segmentation/_chan_vese.py

```diff
@@ -426,15 +426,15 @@ def chan_vese(
         image = image / cp.max(image)
 
     i = 0
+    mu = cp.asarray(mu, dtype=float_dtype)
+    lambda1 = cp.asarray(lambda1, dtype=float_dtype)
+    lambda2 = cp.asarray(lambda2, dtype=float_dtype)
     if extended_output:
         old_energy = _cv_energy(image, phi, mu, lambda1, lambda2)
         energies = []
-    phivar = tol + 1
 
+    phivar = tol + 1
```

Comment:
Think we want to do the same with `tol`

```suggestion
    tol = cp.asarray(tol, dtype=float_dtype)
    phivar = tol + 1
```

Reply:
For `tol`, it is not passed to any fused kernels, but it may still be good to transfer it to device in this way

Reply:
Yeah think it is a good idea to do it with all scalars. It is too easy to miss this kind of thing and there is not much cost to converting them all

Can see you already did this above. So marking resolved

Thanks Greg! 🙏

Reply:
Providing a little more context on why this is helpful for `tol` specifically in the comments review comments below